sync/class.jetpack-sync-functions.php

```diff
@@ -66,7 +66,38 @@ public static function sanitize_taxonomy( $taxonomy ) {
 	public static function get_post_types() {
 		global $wp_post_types;
 
-		return $wp_post_types;
+		$post_types_without_callbacks = array();
+		foreach ( $wp_post_types as $post_type_name => $post_type ) {
+			$sanitized_post_type = self::sanitize_post_type( $post_type );
+			if ( ! empty( $sanitized_post_type ) ) {
+				$post_types_without_callbacks[ $post_type_name ] = $sanitized_post_type;
+			} else {
+				error_log( 'Jetpack: Encountered a recusive post_type:' . $post_type_name );
+			}
+		}
+		return $post_types_without_callbacks;
+	}
+
+	public static function sanitize_post_type( $post_type ) {
+		// Lets clone the post type object instead of modifing the global one.
+		$cloned_post_type = json_decode( wp_json_encode( $post_type ) );
```

Comment:
I think we can skip the cloning part as we are assigning everything to a new array.